include/micro/core/exception.h

```diff
@@ -0,0 +1,21 @@
+#ifndef MICRO_EXCEPTION_H
+#define MICRO_EXCEPTION_H
+
+#include <stdexcept>
+
+namespace micro
+{
+    class MicroException : public std::exception
+    {
+    public:
+        virtual const char *what() const throw();
+    };
+
+    class BufferOverflow : public MicroException
```

Comment:
C++ already has an [exception](http://www.cplusplus.com/reference/stdexcept/overflow_error/) for this type of errors